mira/metamodel/templates.py

```diff
@@ -66,6 +168,58 @@ def from_json(cls, data) -> "Template":
         stmt_cls = getattr(sys.modules[__name__], template_type)
         return stmt_cls(**data)
 
+    def is_equal_to(self, other: "Template", with_context: bool = False) -> bool:
+        if not isinstance(other, Template):
+            return False
+        return templates_equal(self, other, with_context)
+
+    def refinement_of(
+        self,
+        other: "Template",
+        refinement_func: Callable[[str, str], bool],
+        with_context: bool = False,
+    ) -> bool:
+        """Check if this template is a more detailed version of another"""
+        if not isinstance(other, Template):
+            return False
+
+        if self.type != other.type:
+            return False
+
+        for field_name in self.__fields__:
+            this_value = getattr(self, field_name)
+
+            # Skip 'type'
+            if field_name == "type":
+                continue
+
+            # Check refinement for any attribute that is a Concept; this is
+            # strict in the sense that unless every concept of this template is a
+            # refinement of the other, the Template as a whole cannot be
+            # considered a refinement
+            elif isinstance(this_value, Concept):
+                other_concept = getattr(other, field_name)
+                if not this_value.refinement_of(
+                    other_concept, refinement_func=refinement_func, with_context=with_context
+                ):
+                    return False
+
+            elif isinstance(this_value, list):
+                if len(this_value) > 0:
+                    if isinstance(this_value[0], Provenance):
```

Comment:
I don't think `Provenance` plays a role in refinement finding, we don't need to worry about it.

Reply:
Resolved in https://github.com/indralab/mira/pull/19/commits/e8ec13236267ce4f0d5f53f667e66ddd0ae4dd62